examples/fluids/qfunctions/sgs_dd_utils.h

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and other CEED contributors.
+// All Rights Reserved. See the top-level LICENSE and NOTICE files for details.
+//
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This file is part of CEED:  http://github.com/ceed
+
+/// @file
+/// Structs and helper functions for data-driven subgrid-stress modeling
+/// See 'Invariant data-driven subgrid stress modeling in the strain-rate eigenframe for large eddy simulation' 2022 and 'S-frame discrepancy
+/// correction models for data-informed Reynolds stress closure' 2022
+
+#ifndef sgs_dd_utils_h
+#define sgs_dd_utils_h
+
+#include <ceed.h>
+
+#include "newtonian_state.h"
+#include "newtonian_types.h"
+#include "utils.h"
+#include "utils_eigensolver_jacobi.h"
+
+// @brief Calculate the inverse of the multiplicity, reducing to a single component
+CEED_QFUNCTION(InverseMultiplicity)(void *ctx, CeedInt Q, const CeedScalar *const *in, CeedScalar *const *out) {
+  const CeedScalar(*multiplicity)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[0];
+  CeedScalar(*inv_multiplicity)               = (CeedScalar(*))out[0];
+
+  CeedPragmaSIMD for (CeedInt i = 0; i < Q; i++) inv_multiplicity[i] = 1.0 / multiplicity[0][i];
+  return 0;
+}
```

Comment:
It feels like a lot of work to spin up an operator just to invert a vector's contents?

Reply:
I'll go ahead and merge this, but let me know if you thing of a way to get out  a vector of single components.

Reply:
We can probably add this to the interface - I'll think on the best way to do so